source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -612,6 +619,42 @@ def _getAvailableAddonsInBG(self):
 		core.callLater(delay=0, callable=self.detailsVM.updated.notify, addonDetailsVM=self.detailsVM)
 		log.debug("completed refresh")
 
+	def cancelDownloadAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		try:
+			addonDataManager._downloadsPendingCompletion.remove(listItemVM)
+		except KeyError:
+			log.debug("Download already completed")
+		else:
+			futuresCopy = self._downloader._pending.copy()
+			for future, addon in futuresCopy.items():
+				if listItemVM == addon[0] and not future.cancel():
+					self._downloader.progress.pop(addon[0], None)
+		finally:
+			listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
+
+	def cancelInstallAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		log.debug(f"Cancelling install of {listItemVM.Id}")
+
+		if listItemVM in addonDataManager._downloadsPendingCompletion:
+			self.cancelDownloadAddon(listItemVM)
+		elif listItemVM.status == AvailableAddonStatus.DOWNLOAD_SUCCESS:
+			pendingInstallCopy = addonDataManager._downloadsPendingInstall.copy()
+			for addonData, fileDownloaded in pendingInstallCopy:
+				if addonData == listItemVM:
+					addonDataManager._downloadsPendingInstall.remove((addonData, fileDownloaded))
+					# Clean up download file
+					try:
+						os.remove(fileDownloaded)
+					except Exception as e:
+						log.error(f"Failed to delete downloaded file {fileDownloaded}: {e}")
+
+			listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
+
+	def cancelInstallAddons(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]):

```

Comment:
To match other suggestions

```suggestion
	def cancelInstallForAddons(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]):
```

```suggestion
	def cancelAddonsInstall(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]):
```